### PR TITLE
[network/mempool] Improve network mock tests

### PR DESCRIPTION
**Pull request by gregnazario**
## Motivation

This is a followup on the previous PR for network tests

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Existing unit tests in PR

### CODE REVIEW

File: network/src/testutils/test_node.rs [hunk 1]
```diff
@@ -345,3 +365,82 @@ pub fn mock_conn_metadata(
...
+    (peer, metadata)
+}
+
+/// Creates a mock connection based on the `Vfn` to `Validator` connection
+pub fn vfn_validator_mock_connection(
+    origin: ConnectionOrigin,
+    protocol_ids: &[ProtocolId],
+) -> (PeerNetworkId, ConnectionMetadata) {
+    let peer = PeerNetworkId::new(NetworkId::Vfn, PeerId::random());
+    let metadata = match origin {
```
Comment (JoshLind) [inline network/src/testutils/test_node.rs:417]: nit: you could just `match origin` to extract the peer role, right? Everything else is the same? e.g.,

```
let role = match origin {
...
};
let metadata = mock_conn_metadata(peer, role, origin, Some(protocol_ids));
```

File: network/src/testutils/test_node.rs [hunk 2]
```diff
@@ -345,3 +365,82 @@ pub fn mock_conn_metadata(
...
+    (peer, metadata)
+}
+
+/// Creates a mock connection based on the `Pfn` to `Vfn` connection
+pub fn pfn_vfn_mock_connection(
+    origin: ConnectionOrigin,
+    protocol_ids: &[ProtocolId],
+) -> (PeerNetworkId, ConnectionMetadata) {
+    let peer = PeerNetworkId::new(NetworkId::Public, PeerId::random());
+    let metadata = match origin {
```
Comment (JoshLind) [inline network/src/testutils/test_node.rs:435]: nit: Same here regarding just extracting the role.

File: network/src/testutils/test_node.rs [hunk 3]
```diff
@@ -345,3 +365,82 @@ pub fn mock_conn_metadata(
     }
     metadata
 }
+
+/// Creates a mock connection based on the `Validator` to `Validator` connection
+pub fn validator_mock_connection(
```
Comment (JoshLind) [inline network/src/testutils/test_node.rs:402]: nit: you could have all of these methods call a single private helper, e.g.,

```
fn mock_peer_and_metadata(...) -> (PeerNetworkId, ConnectionMetadata) {
    let peer = PeerNetworkId::new(network_id, peer_id);
    let metadata = mock_conn_metadata(peer, peer_role, origin, Some(protocol_ids));
    (peer, metadata)
}
```

File: mempool/src/tests/integration_tests.rs [hunk 4]
```diff
@@ -29,7 +29,7 @@ fn single_node_test() {
         other_peer_network_id,
         PeerRole::Validator,
         ConnectionOrigin::Outbound,
-        Some(&[ProtocolId::MempoolDirectSend]),
+        &[],
```
Comment (JoshLind) [inline mempool/src/tests/integration_tests.rs]: Silly question: why did we remove these? Is it because they're the same as the default protocols?

**Review by JoshLind — APPROVED** [review]
Thanks @gregnazario! LGTM. A couple nits, but it looks cleaner 😄

File: network/src/testutils/test_node.rs (hunk 1 above)
Comment (gregnazario) [reply network/src/testutils/test_node.rs:417]: Yeah, I can fix that.

File: network/src/testutils/test_node.rs (hunk 3 above)
Comment (gregnazario) [reply network/src/testutils/test_node.rs:402]: Yeah, I can do that.  For the couple lines I wasn't super worried

File: mempool/src/tests/integration_tests.rs (hunk 4 above)
Comment (gregnazario) [reply mempool/src/tests/integration_tests.rs]: Yeah, it's the same as the default protocols.  Then when they are changed we don't have to go through and be explicit.

**Review by JoshLind — DISMISSED** [review]
LGTM!

**Pull request closed without merging**